File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

import static java.util.Objects.requireNonNull;

class Gigasecond {

    private static final long GIGA_SECOND = 1_000_000_000;
    private final LocalDateTime birthDateTime;

    Gigasecond(final LocalDate birthDate) {
        requireNonNull(birthDate, "birthDate must not be null");
        birthDateTime = birthDate.atStartOfDay();
    }

    Gigasecond(final LocalDateTime birthDateTime) {
        this.birthDateTime = requireNonNull(birthDateTime, "birthDateTime must not be null");
    }

    LocalDateTime getDateTime() {
        return birthDateTime.plusSeconds(GIGA_SECOND);
    }

}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime


class Gigasecond {
    private val GIGA_SECOND: Long = 1_000_000_000L
    private val localDateTime: LocalDateTime

    constructor(localDate: LocalDate) {
        localDateTime = localDate.atStartOfDay()
    }

    constructor(localDateTime: LocalDateTime) {
        this.localDateTime = localDateTime
    }

    private fun calculateGigaSecond() = localDateTime.plusSeconds(GIGA_SECOND)

    val date: LocalDateTime by lazy<LocalDateTime> { calculateGigaSecond() }
}